app/assets/javascripts/search.js.coffee.erb

```diff
@@ -0,0 +1,28 @@
+class @Search
```

Comment:
it's also weird that you're making this a globally accessible class by attaching it to this.